libkatana_python_native/src/EntityTypeManager.cpp

```diff
@@ -103,5 +120,42 @@ katana::python::InitEntityTypeManager(py::module_& m) {
             } else {
               return py::cast(EntityType{self, id});
             }
-          });
+          })
+      .def(
+          "get_atomic_subtypes",
+          [](const katana::EntityTypeManager* self, EntityType type) {
+            py::set ret;
+            if (self->HasEntityType(type.type_id)) {
+              auto type_set = self->GetAtomicSubtypes(type.type_id);
+              for (EntityTypeID subtype_id = 0; subtype_id < type_set.size();
+                   ++subtype_id) {
+                if (type_set.test(subtype_id)) {
+                  if (self->GetAtomicTypeName(subtype_id).has_value()) {
+                    ret.add(AtomicEntityType{{self, subtype_id}});
+                  }
+                }
+              }
+            }
+            return ret;
+          })
+      .def(
+          "get_supertypes",
+          [](const katana::EntityTypeManager* self, EntityType type) {
+            py::set ret;
+            if (self->GetAtomicTypeName(type.type_id).has_value()) {
+              auto type_set = self->GetSupertypes(type.type_id);
+              for (EntityTypeID subtype_id = 0; subtype_id < type_set.size();
+                   ++subtype_id) {
+                if (type_set.test(subtype_id)) {
+                  ret.add(EntityType{self, subtype_id});
+                }
+              }
+            }
+            return ret;
+          })
+      .def(
+          "get_num_atomic_types", &katana::EntityTypeManager::GetNumAtomicTypes)
+      .def(
+          "get_num_of_entity_types",
+          &katana::EntityTypeManager::GetNumEntityTypes);
```

Comment:
`GetNumEntityTypes` should be called as `num_types` (all types are entity types). `GetNumAtomicTypes` should be called as `num_atomic_types`.

The `Get` at the beginning is an oddity even in C++.